### ISSUE TITLE
simple_job_models.py

### ISSUE THREAD

**Issue by albertma-evotec** [jamesgleave/Deep-Docking-NonAutomated#8]
Hi

I am trying to understand the code because so far it is not working for me. I am still stuck at the first iteration.
I have a DB of 4.8M compounds. I am using a smaller DB because I want to make sure it is working to some extents before committing to a larger DB. I have docked ~24000 compounds (8000 each for training set, validation set and test set)
The training_labels.txt, validation_labels.txt and test_labels.txt files are created to store the docking scores and IDs.

`python phase_2/simple_job_models.py -n_it 1 -mdd green_DB_MorganFP/ -time 00-60:00 -file_path deep-docking-v2/proteinx -nhp 12 -titr 8 -n_mol 8000
`
and ran the 24 scripts generated. For each of them, it firstly says "Training size not specified, using entire dataset..." (I am not sure if this is normal) then always ends with "There are not enough hits... exiting."

![image](https://user-images.githubusercontent.com/54853041/115855482-33838500-a423-11eb-9bd1-1f22d6ae9056.png)

I then changed the n_mol to 100000
python phase_2/simple_job_models.py -n_it 1 -mdd green_DB_MorganFP/ -time 00-60:00 -file_path deep-docking-v2/proteinx -nhp 12 -titr 8 -n_mol 100000
![image](https://user-images.githubusercontent.com/54853041/115855619-5e6dd900-a423-11eb-9ee3-9288d5db9dce.png)
It does not help.

I realised the code may only work for training size = 1M, so I then changed all "1000000" number in progressive_docking.py file to 8000 which is my training set size but it did not help at all. It still complains about not enough hits. But from my understanding there will always be some "hits" based on how many top score compounds I defined them as hits

Apart from this, I actually want to understand this bit

line 75 to 79 in simple_job_models.py

```
# sums the first column and divides it by 1 million 
## is this the average score for the molecules? Why divide by 1000000?
### [[0]] would return a dataframe and .sum() is buggy on dataframes (see: https://stackoverflow.com/questions/52912693/pandas-sum-return-the-infinite-value)
### replace with => pd.to_numeric(df[0]).sum()  # returns a single number representing the sum
t_mol = pd.read_csv(mdd+'/Mol_ct_file.csv',header=None)[[0]].sum()[0]/1000000 # num of compounds in each file is mol_ct_file
```

what is this t_mol suppose to be or play any role in the training?
Can you comment on the comments, does it need to correct?

**Comment by fgentile89** [comment]
Hi, 
it looks like your train and validation sets were not read correctly (their shape is just 0,1). This causes the "not enough hits" problem. Could you check the size of training morgan fingerprints (in morgan folder) and labels (in iteration folder)? Also, did you create those with the provided scripts?

**Comment by jyaacoub** [comment]
Yeah, so t_mol is the database size per million. So a database of 4.8M compounds would have t_mol be equal to 4.8. And from the screenshot, you sent you can see t_mol has a value of around 4.8 (outlined in red below).

![t_mol param](https://user-images.githubusercontent.com/50300488/116744881-ca72b300-a9c8-11eb-8622-a696081b6825.png)

**Comment by albertma-evotec** [comment]
Thanks for the reply. I was confused because I did not get the purpose of divide DB size by 1000000. I still dont get it because 1 million does not mean anything to me (i am not using a 1M dataset neither a 1M training set but thats okay...)

lets assume size of DB should always be divided 1M to calculate t_mol 
![image](https://user-images.githubusercontent.com/54853041/116753818-9c5c9580-a9ff-11eb-9ecc-93f781d95e1b.png)

then t_mol is used in calculating the first_mols if percent_first_mols=-1 (i.e. when user doesn't explicitly assign a value)
![image](https://user-images.githubusercontent.com/54853041/116754563-d8dcc100-aa00-11eb-860a-de2930f87da0.png)

In my run, I am going to set percent_first_mols to be 0.2, so that's fine. And first_mols will be = int(percent_last_mols * len(scores_val)) which make sense to me. And I think I should explicitly set a number regardless which iteration the run is at. I say this because in line 40 in simple_job_models.py
`parser.add_argument('-pfm', '--percent_first_mols', required=False, default=-1, help='Fraction of top scoring molecules to be considered as virtual hits in the first iteration (for standard DD run on 11 iterations, we recommend 0.01)')`
The hint gives me an impression that this percent_first_mols value is only relevant in the first iteration which I don think is true. One should explicitly set percent_first_mols to be something like 0.1 then gradually decreases in the subsequent iterations. For instance, iteration 1, percent_first_mols is 0.1; iteration 2, percent_first_mols is 0.09; iteration 3, percent_first_mols is 0.08 and so on. Am I correct?

**Comment by fgentile89** [comment]
Hi Albert,
the per million thing should work fine, no need to change it. It just tells how many compounds will be sampled per million from each file, but if you specify less sampling molecules it will be rescaled. We used that in an earlier version of the program and we kept it just in case we need in the future. To confirm, just check that your *set.txt files are populated as you expect. 

Regarding the percentage, you just need to specify initial one and final one. It is gradually decreased in the script for later iterations (not directly on the percent value but see line 124

```
good_mol = int(((100-first_mols)*n_it + titr*first_mols-100)/(titr-1))
```

Let us known if this makes sense and if you encounter other issues.

**Comment by BJWiley233** [comment]
How large should the first training, test, and validation set be.  I was just testing the package to see how/if I could get it to work.  After using a threshold to keep docked molecules, I had (not including header line) only 143 molecules for training_labels.txt, 51 for testing_labels.txt , and  47 for validation_labels.txt.  I also get this below.  Also not sure why shape is 1 less molecule for train (see below 142) and test (see below 50)

```
Data acquired...
Train shape: (142, 1) Valid shape: (47, 1) Test shape: (50, 1)
Data Augmentation iteration 1 data shape: (142, 1)
Training labels shape:  (142, 1)
There are not enough hits... exiting.
qqq
